### ISSUE TITLE
Fatal performance error when you stop using the cell phone

### ISSUE THREAD

**Issue by LuisPerss** [wheremyfoodat/Panda3DS#679]
That is to say, when you are using the emulator but you take your fingers off of it, the FPS starts to decrease a lot for some reason, I think it is a serious bug that has to be corrected for future updates, that is to say, from the 50fps that it goes at, for example, if I take my fingers off of the emu, that is, I leave the phone while the game continues, it drops to 20 15 and that is very bad, but when you tap the screen, boom, it returns to its normal state but only if you keep your fingers or hand on the phone, it is an error.

**Comment by LuisPerss** [comment]
I don't think it's Android, since that doesn't happen in Lime3DS or Mandarin3DS. I think it's Panda, and it also happens in Dolphin Wii.
I know what you mean about energy saving, because Samsung's GameLauncher has that option if you leave the phone inactive for 15 seconds, but this emulator is not compatible with GameLauncher or its plugins.

**Comment by LuisPerss** [comment]
The emulator has good performance until you unconsciously leave the screen still or for some other reason you leave it and boom the performance plummets because of what I just mentioned and that cuts the whole experience

**Comment by Ishan09811** [comment]
This is occurring because panda doesn't uses foreground service which keep active the emulation in Lime3ds or mandarine case

**Comment by Ishan09811** [comment]
> @LuisPerss This is occurring because panda doesn't uses foreground service which keep active the emulation like in the Lime3ds or mandarine case

**Comment by GabrielBRDeveloper** [comment]
Hi, so, you phone has adaptive refresh rate display?

**Comment by LuisPerss** [comment]
Indeed, my cell phone has a screen with an adaptive frequency of 60 to 90hz, why?

**Comment by Kvn0l** [comment]
> The emulator has good performance until you unconsciously leave the screen still or for some other reason you leave it and boom the performance plummets because of what I just mentioned and that cuts the whole experience

It's not really emulator fault.
This clock reduction is a power saving "feature" present in many mediatek/exynos (the condition for throttle to happen are different though), its related to thermal profiles (it don't throttle what vendor whitelisted) and governor, it can only be solved by your phone's vendor provider or with root to force freqs at max.
The behavior of ramping up frequencies on touch is called Touch Boost if I'm not mistaken, mtk """magic"""

**Comment by LuisPerss** [comment]
> > El emulador tiene un buen rendimiento hasta que inconscientemente dejas la pantalla quieta o por alguna otra razón lo dejas y boom el rendimiento se desploma por lo que acabo de mencionar y eso corta toda la experiencia
> 
> En realidad no es un fallo del emulador. Esta reducción de reloj es una "función" de ahorro de energía presente en muchos MediaTek/Exynos (aunque las condiciones para que se produzca la limitación son diferentes), está relacionada con los perfiles térmicos (no limita lo que el proveedor ha incluido en la lista blanca) y el regulador, solo puede solucionarlo el proveedor de tu teléfono o con root para forzar las frecuencias al máximo. El comportamiento de aumentar las frecuencias al tocar se llama Touch Boost si no me equivoco, mtk """magic"""

Ok, I understand, it's a MediaTeks thing, but isn't it something that the emulator can fix? I ask from my ignorance, because this doesn't happen in other 3DS emulators, such as lime3ds and the pablomk7 project and mandarin 3ds.

**Comment by Kvn0l** [comment]
> > > El emulador tiene un buen rendimiento hasta que inconscientemente dejas la pantalla quieta o por alguna otra razón lo dejas y boom el rendimiento se desploma por lo que acabo de mencionar y eso corta toda la experiencia
> > 
> > 
> > En realidad no es un fallo del emulador. Esta reducción de reloj es una "función" de ahorro de energía presente en muchos MediaTek/Exynos (aunque las condiciones para que se produzca la limitación son diferentes), está relacionada con los perfiles térmicos (no limita lo que el proveedor ha incluido en la lista blanca) y el regulador, solo puede solucionarlo el proveedor de tu teléfono o con root para forzar las frecuencias al máximo. El comportamiento de aumentar las frecuencias al tocar se llama Touch Boost si no me equivoco, mtk """magic"""
> 
> Ok, I understand, it's a MediaTeks thing, but isn't it something that the emulator can fix? I ask from my ignorance, because this doesn't happen in other 3DS emulators, such as lime3ds and the pablomk7 project and mandarin 3ds.

No, only the manufacturer of the phone can solve it (or yourself with root), only thing Panda's devs can do is flag the app as game, and this is already done by them
Lime3DS and Citra are in manufacturer whitelist/gamelist, what you could also do is edit Panda's apk package name to com.antutu.ABenchMark or even io.github.lime3ds.android and test if you can get any difference (Google/reddit have guide on hows)

**Comment by LuisPerss** [comment]
> > > > El emulador tiene un buen rendimiento hasta que inconscientemente deja la pantalla quieta o por alguna otra razón lo deja y boom el rendimiento se desploma por lo que acabo de mencionar y eso corta toda la experiencia
> > > 
> > > 
> > > En realidad no es un fallo del emulador. Esta reducción de reloj es una "función" de ahorro de energía presente en muchos MediaTek/Exynos (aunque las condiciones para que se produzca la limitación son diferentes), está relacionada con los perfiles térmicos (no limita lo que el proveedor ha incluido en la lista blanca) y el regulador, solo puede solucionarlo el proveedor de tu teléfono o con root para forzar las frecuencias al máximo. El comportamiento de aumentar las frecuencias al tocar se llama Touch Boost si no me equivoco, mtk """magic"""
> > 
> > 
> > Vale, ya entiendo, es cosa de MediaTeks, pero ¿no es algo que el emulador pueda solucionar? Lo pregunto desde mi ignorancia, porque esto no pasa en otros emuladores de 3DS, como lime3ds y el proyecto pablomk7 y mandarin 3ds.
> 
> No, solo el fabricante del teléfono puede resolverlo (o usted mismo con root), lo único que pueden hacer los desarrolladores de Panda es marcar la aplicación como juego, y esto ya lo hacen ellos. Lime3DS y Citra están en la lista blanca/lista de juegos del fabricante, lo que también podría hacer como solución temporal es editar el nombre del paquete apk de Panda a com.antutu.ABenchMark o incluso io.github.lime3ds.android y probar si puede obtener alguna diferencia (Google/reddit tienen una guía sobre cómo hacerlo)

ok, isn't there a possibility for developers to put it as an app? just like lime and mandarine? just like the system detects the apks and gives it system functions, such as the possibility of using the game launcher of your native system or the screen capture?

**Comment by LuisPerss** [comment]
> Lime y Mandarine también están marcados como juegos por sus respectivos desarrolladores, como ya dije, es culpa del fabricante del teléfono/mtk y no es algo que el desarrollador de panda (o cualquier desarrollador de aplicaciones) pueda resolver.

I understand, thanks for answering
